src/video/windows/SDL_windowsevents.c

```diff
@@ -861,29 +861,54 @@ static bool HasDeviceID(Uint32 deviceID, const Uint32 *list, int count)
 }
 
 #if !defined(SDL_PLATFORM_XBOXONE) && !defined(SDL_PLATFORM_XBOXSERIES)
-static void GetDeviceName(HDEVINFO devinfo, const char *instance, char *name, size_t len)
+static void GetDeviceName(HDEVINFO devinfo, const char *devName, const char *instance, char *name, size_t len)
 {
     name[0] = '\0';
+    name[len-1] = '\0';
 
-    SP_DEVINFO_DATA data;
-    SDL_zero(data);
-    data.cbSize = sizeof(data);
-    for (DWORD i = 0;; ++i) {
-        if (!SetupDiEnumDeviceInfo(devinfo, i, &data)) {
-            if (GetLastError() == ERROR_NO_MORE_ITEMS) {
+    HANDLE hFile = CreateFileA(devName, GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE, NULL, OPEN_EXISTING, 0, NULL);
+    if (hFile != INVALID_HANDLE_VALUE) {
+        char *product_string = NULL;
+        WCHAR string[128];
+
+        if (SDL_HidD_GetProductString(hFile, string, sizeof(string))) {
+            product_string = WIN_StringToUTF8W(string);
+        }
+
+        for (size_t i = 0; i < len-1; i++) {
+            name[i] = product_string[i];
```

Comment:
This will crash if product_string is NULL. It also might read more characters than product_string has.